src/content/interop/js-interop/js-types.md

```diff
@@ -38,6 +38,15 @@ JS types form a natural type hierarchy:
 
 You can find the definition of each type in the [`dart:js_interop` API docs].
 
```

Comment:
I think the new content should either be 1) primarily in the api docs, or 2) in a breakout section on this page.

It's a sizable amount of digression for the Type hierarchy section. I think either of these options would also allow more room for Siggi's [comment](https://github.com/dart-lang/site-www/pull/5668/files#r1540041570) about elaborating on when to use each.

1) API docs:

Add a sub-bullet under the `JSBoxedDartObject` item in the list (line 38) and put your first two sentences there (slight rewrite for wording):

```markdown
- `JSBoxedDartObject`, which allows users to box and pass Dart values
    opaquely within the same Dart runtime.
  - From Dart 3.4 onwards, the type `ExternalDartReference` in `dart:js_interop` can be used to
    accomplish the same thing, but is *not* a JS type. To learn more about the tradeoffs between
    each option, read [API docs link].
```

And then your second paragraph (lines 44-48) could be a note in the API docs for either or both types (I think both makes sense).

2) Another section on this page:

Add the same sub-bullet in 1), but link to the section on the page instead (`To learn more about the tradeoffs between each option, read [link to section]`).

At the bottom of the page, add a section and include your second paragraph + Siggi's "when a JS type is needed" explanation:

```markdown
## `JSBoxedDartObject` vs `ExternalDartReference`

Both `JSBoxedDartObject` and `ExternalDartReference` pass opaque references to
Dart `Object`s through JavaScript. However, `JSBoxedDartObject` wraps the opaque
reference in a JavaScript object, while `ExternalDartReference` is the reference
itself and therefore is not a JS type. Use `ExternalDartReference` where
performance is needed and `JSBoxedDartObject` where a JS type is needed.

(edit as needed: )
For example, if you just want to pass references opaquely back and forth using the
reference as an identity, then `ExternalDartReference` is a good fit. However, if you want
a reference that can also store data and you want to extend it to add data to that reference,
then we recommend using `JSBoxedDartObject`. 
```

And I'd even add more of the tradeoff details from your issue https://github.com/dart-lang/sdk/issues/55187

---

I don't have a preference, though I'm not too familiar with API doc best practices so I'm not sure if 1) makes sense for the kind of information usually kept there, or if repeating the explanation on both type's pages is bad practice 🤷‍♀️

Reply:
Thanks for the suggestion! I like the idea of splitting it into a separate section (2) and avoiding polluting the type hierarchy section so I went ahead and did that.